deployment_scripts/linode-marketplace-valkey/valkey-deploy.sh

```diff
@@ -0,0 +1,121 @@
+# <UDF name="state_or_province_name" label="State or Province" example="Example: Pennsylvania" />
+# <UDF name="locality_name" label="Locality" example="Example: Philadelphia" />
+# <UDF name="organization_name" label="Organization" example="Example: Akamai Technologies" />
+# <UDF name="email_address" label="Email Address" example="Example: webmaster@example.com" />
+# <UDF name="ca_common_name" label="CA Common Name" example="Example: Valkey CA" />
+
+## Valkey Version. Valkey is in early access, with only one stable version, and no beta branch established. 
+#<UDF name="valkey_version" label="Which version of Valkey to install" oneOf="7.2.5">
+# Client Count
+# <UDF name="client_count" label="Number of clients connecting to Valkey">
+
+# git repo
+export GIT_REPO="https://github.com/akamai-compute-marketplace/marketplace-apps.git"
+export WORK_DIR="/tmp/marketplace-apps" 
+export MARKETPLACE_APP="apps/linode-marketplace-valkey"
+# export BRANCH=""
+
+# enable logging
+exec > >(tee /dev/ttyS0 /var/log/stackscript.log) 2>&1
+
+function cleanup {
+  if [ -d "${WORK_DIR}" ]; then
+    rm -rf ${WORK_DIR}
+  fi
+}
+
+function udf {
+  
+  local group_vars="${WORK_DIR}/${MARKETPLACE_APP}/group_vars/linode/vars"
+  sed 's/  //g' <<EOF > ${group_vars}
+
+  # sudo username
+  username: ${USER_NAME}
+  # valkey version
+  valkey_version: ${VALKEY_VERSION}
+  # client count
+  client_count: ${CLIENT_COUNT}
+  # ssl vars
+  country_name: ${COUNTRY_NAME}
+  state_or_province_name: ${STATE_OR_PROVINCE_NAME}
+  locality_name: ${LOCALITY_NAME}
+  organization_name: ${ORGANIZATION_NAME}
+  email_address: ${EMAIL_ADDRESS}
+  ca_common_name: ${CA_COMMON_NAME}
+EOF
+
+  if [ "$DISABLE_ROOT" = "Yes" ]; then
+    echo "disable_root: yes" >> ${group_vars};
+  else echo "Leaving root login enabled";
+  fi
+
+  # domain vars
+  if [[ -n ${SOA_EMAIL_ADDRESS} ]]; then
+    echo "soa_email_address: ${SOA_EMAIL_ADDRESS}" >> ${group_vars};
+  fi
+
+  if [[ -n ${DOMAIN} ]]; then
+    echo "domain: ${DOMAIN}" >> ${group_vars};
+  else
+    echo "default_dns: $(hostname -I | awk '{print $1}'| tr '.' '-' | awk {'print $1 ".ip.linodeusercontent.com"'})" >> ${group_vars};
+  fi
+
+  if [[ -n ${SUBDOMAIN} ]]; then
+    echo "subdomain: ${SUBDOMAIN}" >> ${group_vars};
+  else echo "subdomain: www" >> ${group_vars};
+  fi
+
+  if [[ -n ${TOKEN_PASSWORD} ]]; then
+    echo "token_password: ${TOKEN_PASSWORD}" >> ${group_vars};
+  else echo "No API token entered";
+  fi
```

Comment:
```suggestion
```
We are not collecting this in the frontend. This will cause the deploy to fail because it expects it.